src/View/HTML.php

```diff
@@ -68,12 +68,21 @@ public static function createTag($tag, $attributes, $content = null)
             $whitelisted = in_array($attributeKey, $legalEmptyAttributes ?? []);
 
             // Only set non-empty strings (ensures strlen(0) > 0)
+            // Note that boolean `true` passes this condition but boolean `false` does not
+            // This allows setting value-less boolean attributes such as `defer`
             if (strlen($attributeValue ?? '') > 0 || $whitelisted) {
-                $preparedAttributes .= sprintf(
-                    ' %s="%s"',
-                    $attributeKey,
-                    Convert::raw2att($attributeValue)
-                );
+                if ($attributeValue === true) {
```

Comment:
This seems like it has low-risk of being breaking. Currently:

```php
echo SilverStripe\View\HTML::createTag('div', ['aria-expanded' => true]);die;
// <div aria-expanded="1"></div>
```

i.e. it's not rendering as `<div aria-expanded="true"></div>`

As far as I know there aren't any any HTML attributes where Silverstripe users would be passing in a value of `true` and relying on it outputting `"1"`.

I think just a changelog entry here to mention this behavioural change is good enough - I've made a suggestion change in the docs PR